perf: monitor binary size change in PRs via cargo-bloat workflow

lmmx opened the pull request:
- As discussed in #917 

Cargo bloat workflow to show the binary size change for PRs in the form of a "Checks" status line

> I came up with a way that shows it in the PR via a check "status" line, see [demo](https://github.com/lmmx/cargo-bloat-action-demo/pull/1) PR and [workflow source](https://github.com/lmmx/cargo-bloat-action-demo/blob/master/.github/workflows/bloat.yml)
> 
> <img width="1523" height="716" alt="Image" src="https://github.com/user-attachments/assets/69c245ce-f529-4ae0-8f98-3d87c0d9a2dd" /> 

 _Originally posted by @lmmx in [#917](https://github.com/j178/prek/issues/917#issuecomment-3418562809)_

I tried it with a status like this, (to save you having to click through to read the result) but it looks like that won't be allowed to run on forks (i.e. only j178's own PRs to this repo would have it, everyone else's would error) so I removed that and now you have to click through to the step summary to see it:

(...)

To see the result you click through to the step summary where it shows up as an "annotation" at the top and a full report below

- [x] Added mold linker and caching so (initial run took 5m)
- [x] Removed check status line (fails for PRs from other repos)
- [x] Rewrote size change calculation in Python

j178 commented on .github/workflows/bloat.yml:
Is it possible to move this action into `ci.yml`?

lmmx replied:
Yeah sure :+1:

j178 commented on .github/workflows/bloat.yml:
```diff
@@ -0,0 +1,137 @@
+      - name: Compare bloat results
+        id: compare
+        if: github.event_name == 'pull_request'
+        run: |
```
What do you think about implementing this shell script as a Python script instead? I think that might be easier to maintain.

lmmx replied:
Sure yeah, a Python script in `.github/scripts/` would work

lmmx replied:
Marked as resolved but left a comment below

j178 commented on .github/workflows/bloat.yml:
```diff
@@ -0,0 +1,137 @@
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+      - name: Display results (PR)
```
I noticed the “Update commit status” step is missing from this workflow - why is that?

```
      - name: Update commit status
        if: github.event_name == 'pull_request'
        uses: actions/github-script@v7
        with:
          script: |
            await github.rest.repos.createCommitStatus({
              owner: context.repo.owner,
              repo: context.repo.repo,
              sha: context.payload.pull_request.head.sha,
              state: 'success',
              context: 'Binary Size Change',
              description: `${{ steps.compare.outputs.pct_change }} (.text: ${{ steps.compare.outputs.base_size }} → ${{ steps.compare.outputs.pr_size }})`,
              target_url: `${context.serverUrl}/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`
            });
```

lmmx replied:
I took it out as it wouldn't work for PRs from other repos, added a note in the README of the demo

> [!TIP]
> If you primarily accept PRs from forks, delete the "Update commit status" step from the workflow to avoid the skipped check.

Nice idea but suspect it wouldn't actually work for what we wanted it for :-(

j178 commented:
This looks great!

<img width="1051" height="531" alt="image" src="https://github.com/user-attachments/assets/f550fd4b-d0f1-4669-b211-b2d51497bcb5" />

lmmx commented:
I’ve gone and rebased the branches I had open and added their outputs (the annotation section at the bottom of the Summary page with the 🏠 icon)

![IMG_4602](https://github.com/user-attachments/assets/20436516-2de5-4914-88c4-6937938c1301)

and pasted them into the PR description like this

> [Binary Size Change](https://github.com/j178/prek/actions/runs/18629114297/job/53111274368#step:9:312)
> +0.63% (.text: 16.0 MiB → 16.1 MiB)
